### PR TITLE
refactor: break out RegionRequestHandler class from app.py

### PR DESCRIPTION
**Pull request by drduhe**
**Issue #, if available:** n/a

### Notes
This PR introduces refactoring of the `RegionRequestHandler` class to decouple region processing logic from `ModelRunner`. The `RegionRequestHandler` now makes it the region processing logic self-contained, modular, and easier to test. Additionally, changes to the `ModelRunner` class were made to initialize this handler with the appropriate dependencies.

Unit tests were also updated to split out region processing testing associated with the new class.

#### Key Changes:

1. **Decoupled `RegionRequestHandler` from `ModelRunner`:**
   - `RegionRequestHandler` now takes in its own initialization parameters:
     - `region_request_table`, `job_table`, `region_status_monitor`, `endpoint_statistics_table`
     - `tiling_strategy`, `region_request_queue`, `endpoint_utils`, `config`
   - This similarly decouples it from `ModelRunner`, allowing for cleaner design and unit testing.

### Checklist

Before you submit a pull request, please make sure you have the following:
- [x] Code changes are compact and well-structured to facilitate easy review
- [x] Changes are documented in the README.md and other relevant documentation pages
- [x] PR title and description accurately reflect the changes and are detailed enough for historical tracking
- [x] PR contains tests that cover all new code and the code has been manual tested
- [x] All new dependencies are declared (if any), and no unnecessary libraries are added
- [x] Performance impacts (if any) of the changes are evaluated and documented
- [x] Security implications of the changes (if any) are reviewed and addressed
- [x] I have read the [Contributing Guidelines](../CONTRIBUTING.md) and agree to follow the [Code of Conduct](../CODE_OF_CONDUCT.md)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/aws/osml/model_runner/test_reqion_request_handler.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
+#  Copyright 2023-2024 Amazon.com, Inc. or its affiliates.
```
Comment (jtblack-aws) [inline test/aws/osml/model_runner/test_reqion_request_handler.py:1]: blank line after copyright for consistency

File: src/aws/osml/model_runner/region_request_handler.py [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+from .queue import RequestQueue
+from .status import RegionStatusMonitor
+from .tile_worker import TilingStrategy, process_tiles, setup_tile_workers
+
+# Set up logging configuration
+logger = logging.getLogger(__name__)
+
+# GDAL 4.0 will begin using exceptions as the default; at this point the software is written to assume
+# no exceptions so we call this explicitly until the software can be updated to match.
+gdal.UseExceptions()
```
Comment (jtblack-aws) [inline src/aws/osml/model_runner/region_request_handler.py:27]: This seems out of place here.  If we only need to call this once I feel like it should be with app.py still, maybe in init.

File: src/aws/osml/model_runner/region_request_handler.py [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    """
+
+    def __init__(
+        self,
+        region_request_table: RegionRequestTable,
+        job_table: JobTable,
+        region_status_monitor: RegionStatusMonitor,
+        endpoint_statistics_table: EndpointStatisticsTable,
+        tiling_strategy: TilingStrategy,
+        region_request_queue: RequestQueue,
```
Comment (edparris) [inline src/aws/osml/model_runner/region_request_handler.py:42]: I don't think we need access to the region request queue in the region handler.

File: src/aws/osml/model_runner/app.py [hunk 4]
```diff
@@ -881,7 +759,7 @@ def add_properties_to_features(self, image_request_item: JobItem, features: List
         return features
 
     @staticmethod
-    def get_inference_metadata_property(image_request_item: JobItem, inference_time: str) -> Dict[str, Any]:
+    def w(image_request_item: JobItem, inference_time: str) -> Dict[str, Any]:
```
Comment (jtblack-aws) [inline src/aws/osml/model_runner/app.py:762]: typo?

**Review by edparris — APPROVED** [review]

**Pull request merged**